### ISSUE TITLE
Missing check for `Array.Length` in Index/Range indexer on array

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#82756]
The following scenario should report a diagnostic for missing member `Array.Length` although the codegen doesn't end up depending on it.
This issue is referenced in test
```
    [Fact]
    public void TODO2()
    {
        var src = """
public static class E
{
    public static void Main()
    {
        int[] i = [1, 2, 3];
        _ = i[^1];
        _ = i[1..^1];
    }
}
""";

        var comp = CreateCompilation(src, targetFramework: TargetFramework.Net100);
        comp.MakeMemberMissing(SpecialMember.System_Array__Length);
        comp.VerifyEmitDiagnostics();
    }
```

We'd fixed a similar issue for list-patterns ([PR](https://github.com/dotnet/roslyn/pull/56287)) but the indexer scenario

**Comment by AlekseyTs** [comment]
Compiler can figure out length of an array without a Length property. The behavior is likely By Design. In other places language didn't special cases arrays and, therefore, require a property

**Comment by jcouv** [comment]
Keeping the issue open for now. Let's discuss to confirm expectations
